mm2src/coins/solana/solana_common.rs

```diff
@@ -87,18 +92,33 @@ pub async fn check_sufficient_balance<T>(
     coin: &T,
     max: bool,
     amount: BigDecimal,
-) -> Result<(BigDecimal, BigDecimal), MmError<SufficientBalanceError>>
+    fees: u64,
+) -> Result<(BigDecimal, BigDecimal, BigDecimal), MmError<SufficientBalanceError>>
 where
     T: SolanaCommonOps + MarketCoinOps,
 {
+    let base_balance = coin.base_coin_balance().compat().await?;
+    let sol_required = lamports_to_sol(fees);
+    if base_balance < sol_required {
+        return MmError::err(SufficientBalanceError::AmountTooLow {
+            amount: base_balance.clone(),
+            threshold: &sol_required - &base_balance,
```

Comment:
Please note `threshold` is assumed to be a minimal value since we can generate transaction, so in this case it should be equal to `sol_required`.